build-static.sh

```diff
@@ -117,88 +130,87 @@ if [ -n "${CLEAN}" ]; then
-		cd static-php-cli/
-		git pull
-		composer install --no-dev -a
-		spcCommand="./bin/spc"
-	else
-		git clone --depth 1 https://github.com/crazywhalecc/static-php-cli --branch main
-		cd static-php-cli/
-		composer install --no-dev -a
-		spcCommand="./bin/spc"
+	if [ -n "${RELEASE}" ] && ! type "gh" >/dev/null 2>&1; then
+		packages="${packages} gh"
 	fi
 
-	# extensions to build
-	if [ -z "${PHP_EXTENSIONS}" ]; then
-		# enable EMBED mode, first check if project has dumped extensions
-		if [ -n "${EMBED}" ] && [ -f "${EMBED}/composer.json" ] && [ -f "${EMBED}/composer.lock" ] && [ -f "${EMBED}/vendor/installed.json" ]; then
-			cd "${EMBED}"
-			# read the extensions using spc dump-extensions
-			PHP_EXTENSIONS=$(${spcCommand} dump-extensions "${EMBED}" --format=text --no-dev --no-ext-output="${defaultExtensions}")
-		else
-			PHP_EXTENSIONS="${defaultExtensions}"
-		fi
+	if [ -n "${packages}" ]; then
+		# shellcheck disable=SC2086
+		brew install --formula --quiet ${packages}
 	fi
-	# additional libs to build
-	if [ -z "${PHP_EXTENSION_LIBS}" ]; then
-		PHP_EXTENSION_LIBS="${defaultExtensionLibs}"
+fi
+
+if [ "${SPC_REL_TYPE}" = "binary" ]; then
+	mkdir -p static-php-cli/
+	cd static-php-cli/
+	if [[ "${arch}" =~ "arm" ]]; then
+		dl_arch="aarch64"
+	else
+		dl_arch="${arch}"
 	fi
-	# The Brotli library must always be built as it is required by http://github.com/dunglas/caddy-cbrotli
-	if ! echo "${PHP_EXTENSION_LIBS}" | grep -q "\bbrotli\b"; then
-		PHP_EXTENSION_LIBS="${PHP_EXTENSION_LIBS},brotli"
+	curl -o spc -fsSL "https://dl.static-php.dev/static-php-cli/spc-bin/nightly/spc-linux-${dl_arch}"
+	chmod +x spc
+	spcCommand="./spc"
+elif [ -d "static-php-cli/src" ]; then
+	cd static-php-cli/
+	git pull
+	composer install --no-dev -a --no-interaction
+	spcCommand="./bin/spc"
+else
+	git clone --depth 1 https://github.com/crazywhalecc/static-php-cli --branch main
+	cd static-php-cli/
+	composer install --no-dev -a --no-interaction
+	spcCommand="./bin/spc"
+fi
+
+# extensions to build
+if [ -z "${PHP_EXTENSIONS}" ]; then
+	# enable EMBED mode, first check if project has dumped extensions
+	if [ -n "${EMBED}" ] && [ -f "${EMBED}/composer.json" ] && [ -f "${EMBED}/composer.lock" ] && [ -f "${EMBED}/vendor/installed.json" ]; then
+		cd "${EMBED}"
+		# read the extensions using spc dump-extensions
+		PHP_EXTENSIONS=$(${spcCommand} dump-extensions "${EMBED}" --format=text --no-dev --no-ext-output="${defaultExtensions}")
+	else
+		PHP_EXTENSIONS="${defaultExtensions}"
 	fi
-	# The mimalloc library must be built if MIMALLOC is true
-	if [ -n "${MIMALLOC}" ]; then
-		if ! echo "${PHP_EXTENSION_LIBS}" | grep -q "\bmimalloc\b"; then
-			PHP_EXTENSION_LIBS="${PHP_EXTENSION_LIBS},mimalloc"
-		fi
+fi
+# additional libs to build
+if [ -z "${PHP_EXTENSION_LIBS}" ]; then
+	PHP_EXTENSION_LIBS="${defaultExtensionLibs}"
+fi
+# The Brotli library must always be built as it is required by http://github.com/dunglas/caddy-cbrotli
```

Comment:
We provide a Go build tag to disable Brotli support (`nobrotli`). We could add new env var to disable that in a future PR if there is a use case.

Reply:
Definitely a possibility, but I don't see a strong reason to build without brotli support.